.circleci/config.yml

```diff
@@ -26,14 +26,43 @@ jobs:
           venv/bin/pip install --progress-bar off --upgrade -r requirements.txt
     - run:
         name: PyTest
-        command: venv/bin/pytest --black --flake8
+        command: venv/bin/pytest --ignore=pensieve/tests/integration/ 
     - run:
         name: Mypy
         command: venv/bin/mypy pensieve
     - save_cache:
         paths:
         - venv/
         key: *cache_key
+  integration:
+    docker:
+    - image: python:3.8
+    steps:
+    - checkout
+    - skip_forked_pr:
+      run:
+        name: Early return if this build is from a forked PR
+        command: |
+          if [ -n "$CIRCLE_PR_NUMBER" ]; then
+            echo "Cannot pass creds to forked PRs, so marking this step successful"
+            circleci step halt
+          fi
+    - build:
+      run:
+        name: Build
+        command: |
+          python3.8 -m venv venv/
+          venv/bin/pip install --progress-bar off --upgrade -r requirements.txt
+    - pytest_integration_test:
+      run:
+        name: PyTest Integration Test
+        # Google's client libraries will check for GOOGLE_APPLICATION_CREDENTIALS
+        # and use a file in that location for credentials if present;
+        # See https://cloud.google.com/docs/authentication/production
+        command: |
+          export GOOGLE_APPLICATION_CREDENTIALS="/tmp/gcp.json"
+          echo "$GCLOUD_SERVICE_KEY" > "$GOOGLE_APPLICATION_CREDENTIALS"
+          venv/bin/pytest pensieve/tests/integration/
```

Comment:
I have tried using a pytest marker for this, however for some reason it gets ignored. As a workaround, I moved all test that would require GCP authentication to a separate directory. This way it is easier for folks to ignore running these tests locally if they are not logged in to GCP.

Reply:
How do you mean that it gets ignored? Markers don't change what executes unless you specifically include or exclude them.

Reply:
Right, it should work by adding an option `-m integration` or `-m "not integration"`, but even if I set these options, all test still get executed.